recipes/litehtml/all/conanfile.py

```diff
@@ -0,0 +1,138 @@
+import os
+import textwrap
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+
+class LitehtmlConan(ConanFile):
+    name = "litehtml"
+    description = "litehtml is the lightweight HTML rendering engine with CSS2/CSS3 support."
+    license = "BSD-3-Clause"
+    topics = ("render engine", "html", "parser")
+    homepage = "https://github.com/litehtml/litehtml"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "utf8": [True, False],
+        "with_icu": [True, False],
+        "external_gumbo": [True, False],
+        "external_xxd": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "utf8": False,
+        "with_icu": False,
+        "external_gumbo": False,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _min_cppstd(self):
+        return "11"
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._min_cppstd)
+        if self.options.shared and self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("litehtml must be built as a static library on windows")
+
+    def requirements(self):
+        if self.options.with_icu:
+            self.requires("icu/69.1")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.external_xxd == None:
+            if self.settings.os == "Windows":
+                self.options.external_xxd = False
+            else:
+                self.options.external_xxd = True
```

Comment:
Setting default options belongs in `config_options`.
`config_options` is executed before user options are applied.
```suggestion
```

Reply:
Hmm I think that docs need to be updated then
https://docs.conan.io/en/latest/reference/conanfile/attributes.html#default-options

Reply:
The docs are correct:
> You can also set the options conditionally to a final value with configure() instead of using default_options:

But it doesn't say that it will override a user option.

[This section does](https://docs.conan.io/en/latest/reference/conanfile/methods.html#configure-config-options):

> These methods can also be used to assign values to options as seen in options. Values assigned in the configure() method cannot be overriden, while values assigned in config_options() can.

Reply:
But as my latest comments, the gumbo and xxd options can/should be removed.
As xxd should be used always, to support cross building on all platforms
And gumbo is used always. Right now, a vendored library is used where we might want to use a cci package in the future (hence the fixme).